ngraph/frontend/paddlepaddle/src/model.cpp

```diff
@@ -210,35 +273,24 @@ namespace ngraph
             }
         }
 
-        InputModelPDPD::InputModelPDPDImpl::InputModelPDPDImpl(const std::string& path,
+        template <typename T>
+        InputModelPDPD::InputModelPDPDImpl::InputModelPDPDImpl(const std::basic_string<T>& path,
                                                                const InputModel& input_model)
             : m_fw_ptr{std::make_shared<ProgramDesc>()}
             , m_input_model(input_model)
         {
-            std::string ext = ".pdmodel";
-            std::string model_file(path);
-            std::unique_ptr<std::ifstream> weights_stream;
-            if (model_file.length() >= ext.length() &&
-                (0 == model_file.compare(model_file.length() - ext.length(), ext.length(), ext)))
-            {
-                std::string weights_file(path);
-                weights_file.replace(weights_file.size() - ext.size(), ext.size(), ".pdiparams");
-                weights_stream = std::unique_ptr<std::ifstream>(
-                    new std::ifstream(weights_file, std::ios::binary));
-                // Don't throw error if file isn't opened
-                // It may mean that model don't have constants
-            }
-            else
-            {
-                model_file += "/__model__";
-            }
+            std::string empty_str = "";
+            std::ifstream weights_stream;
+            std::ifstream pb_stream(pdpd::get_model_path<T>(path, &weights_stream),
+                                    std::ios::in | std::ifstream::binary);
 
-            std::ifstream pb_stream(model_file, std::ios::binary);
+            FRONT_END_GENERAL_CHECK(pb_stream && pb_stream.is_open(), "Model file doesn't exist");
             FRONT_END_GENERAL_CHECK(m_fw_ptr->ParseFromIstream(&pb_stream),
                                     "Model can't be parsed");
 
             loadPlaces();
-            loadConsts(weights_stream ? "" : path, weights_stream.get());
+            loadConsts(weights_stream && weights_stream.is_open() ? std::basic_string<T>{} : path,
```

Comment:
@nosovmik @mvafin  This code only works with pdmodel & pdiparam files. However, there are legacy ` __model__` model files. If using `__model__` The weights_stream is actually not valid.
```suggestion
            loadConsts(weights_stream && weights_stream.is_open() ? std::basic_string<T>{} : path,
                       weights_stream && weights_stream.is_open() ? &weights_stream : nullptr);
```